### PR TITLE
Retain task progress

### LINKED ISSUE #20: Retain task progress across deploys
Use named volumes to store BOINC data to ensure progress towards tasks is maintained as much as possible.

### LINKED ISSUE #23: Missing or incorrect account key
The email address on the Rosetta team was changed which has invalidated the account key. 
Screenshot 2020-04-21 at 16 12 53

### PR DESCRIPTION
**Pull request by chrisys**

### CODE REVIEW

**Comment by chrisys** [comment]
@balena-ci retest

**Review by tmigone — APPROVED** [review]

File: boinc-client/start.sh [hunk 1]
```diff
@@ -1,4 +1,7 @@
 #!/bin/bash
+mkdir -p /usr/app/boinc/locale
+mkdir -p /usr/app/boinc/slots
+
 if [[ -z $ACCOUNT_KEY ]]; then
   echo 'Account key undefined - using balena key'
 else
```
Comment (ptrm) [inline boinc-client/start.sh:7]: The path in `sed` below should be `/usr/app/`**~lib~boinc/**`account_boinc.rosetta.bakerlab.org.xml` as per new [Dockerfile.template](https://github.com/balenalabs/rosetta-at-home/pull/24/files#diff-544f4a594bf9c265a7933f12d8299987R7) from this branch.

Comment (ptrm) [reply boinc-client/start.sh:7]: Also, with persistent storage I have now two account configs:
```
root@ed***:/usr/app/boinc# ls -l account*xml
-rw-r--r-- 1 root root 1195 Apr 21 20:22 account_boinc.bakerlab.org_rosetta.xml
-rw-rw-r-- 1 root root  161 Apr 22 06:48 account_boinc.rosetta.bakerlab.org.xml
root@ed***:/usr/app/boinc# grep auth account*xml
account_boinc.bakerlab.org_rosetta.xml:    <authenticator>208***</authenticator>
account_boinc.rosetta.bakerlab.org.xml:    <authenticator>214***</authenticator>
root@ed***:/usr/app/boinc#
```
The second with balena weak token is used

EDIT:
Actually, the token and user id propagated to more files, and now includes query strings:
```
root@99f*:/usr/app/boinc# grep auth *rosetta*
account_boinc.bakerlab.org_rosetta.xml:    <authenticator>214*</authenticator>
account_boinc.rosetta.bakerlab.org.xml:    <authenticator>214*</authenticator>
sched_reply_boinc.bakerlab.org_rosetta.xml:        <url>http://boinc.bakerlab.org/rosetta/notices.php?userid=2085224&auth=208*</url>
sched_request_boinc.bakerlab.org_rosetta.xml:    <authenticator>208*</authenticator>

root@99f***:/usr/app/boinc# grep 2085224 *
account_boinc.bakerlab.org_rosetta.xml:        <url>http://boinc.bakerlab.org/rosetta/results.php?userid=2085224</url>
client_state.xml:    <userid>2085224</userid>
client_state_prev.xml:    <userid>2085224</userid>
```
```

Comment (chrisys) [reply boinc-client/start.sh:7]: @ptrm good catch, thanks! I think perhaps the original account_boinc.rosetta.bakerlab.org.xml is named incorrectly (if the filename is supposed to match the project URL), and the client is creating the second one. I have this on my devices even running the current version.

Comment (chrisys) [reply boinc-client/start.sh:7]: I'm not sure if updating it in the other files is critical, the client should handle it, but let's monitor it.

Comment (ptrm) [reply boinc-client/start.sh:7]: Thanks. Will the consistency among all other state and scheduled reply xmls be maintained automatically by boinc when the account file fix is applied? I tried fiddling with key replacement in both account files, but got messages about invalid auth when contacting the server, and other xmls remained intact.

One might always change the Rosetta account in webui, but I am concerned about the balena evnironment variable change consequences, especially that  it is easier to propagate the auth via balena itself.

Comment (chrisys) [reply boinc-client/start.sh:7]: @ptrm I've got a couple of test devices going to try this right now, and will create an issue if I see problems - would you mind doing the same if it continues to be a problem on your end too?

**Pull request merged**